python_language/inventory/inventory.py

```diff
@@ -0,0 +1,46 @@
+"""
+Inventory
+---------
+
+Calculate and report the current inventory in a warehouse.
+
+Assume the warehouse is initially empty.
+
+The string, warehouse_log, is a stream of deliveries to
+and shipments from a warehouse.  Each line represents
+a single transaction for a part with the number of
+parts delivered or shipped.  It has the form::
+
+    part_id count
+
+If "count" is positive, then it is a delivery to the
+warehouse. If it is negative, it is a shipment from
+the warehouse.
+
+See :ref:`inventory-solution`.
+"""
+
+warehouse_log = """ frombicator      10
+                    whitzlegidget    5
+                    splatzleblock    12
+                    frombicator      -3
+                    frombicator      20
+                    foozalator       40
+                    whitzlegidget    -4
+                    splatzleblock    -8
+                """
+                
+warehouse_log=warehouse_log.strip()
+list_log=warehouse_log.split('\n')
+log={}
+for x in list_log:
+    k,v=x.split()
+    try:
+        log[k]=int(v)+log[k]
+    except:
```

Comment:
see prior comment about the bare `except` statement here.

Better yet, you can test for membership using `if k in log` or using the `k.get(k, 0)` method, which allows you to specify a default value.